docs/proposals/extensions.md

```diff
@@ -0,0 +1,88 @@
+## Proposal for extending an OpenCompose file with another OpenCompose file
+
+##### Scenario:
+
+Let's say, we have a sample OpenCompose file defining a Wordpress application -
+
+_wordpress-opencompose.yml_ -
+```yaml
+version: '0.1-dev'
+
+services:
+- name: database
+  containers:
+  - image: mariadb:10
+    env:
+    - name: MYSQL_ROOT_PASSWORD
+      value: rootpasswd
+    - name: MYSQL_DATABASE
+      value: wordpress
+    - name: MYSQL_USER
+      value: wordpress
+    - name: MYSQL_PASSWORD
+      value: wordpress
+    ports:
+    - port: 3306
+
+- name: web
+  containers:
+  - image: wordpress:4
+    env:
+    - name: WORDPRESS_DB_HOST
+      value: database:3306
+    - name: WORDPRESS_DB_PASSWORD
+      value: wordpress
+    - name: WORDPRESS_DB_USER
+      value: wordpress
+    - name: WORDPRESS_DB_NAME
+      value: wordpress
+    ports:
+    - port: 80
+      type: external
+```
+
+This file works fine for a developer, but since there is no persistent volumes defined here, we need to alter this for production use cases; but changing the original file disrupts the developer workflow.
+
+So, the production folks create another OpenCompose file which will extend _wordpress-opencompose.yml_ which will look like -
+
+_wordpress.extension.yml_
+
+```yaml
+version: '0.1-dev'
+extends: wordpress-opencompose.yml
```

Comment:
A user can also use the `-f` flag to pass multiple valid OpenCompose files together, right? So, how do we keep track that which extension file extends what?

Also, since all the files are being passed in the same `-f` flag, then if we want one extension file to extend multiple OpenCompose files, how do we do that?

Another approach could be, that the we store the order in which the OpenCompose files are being passed, and the the extension file will extend the one specified just before that? I'm not very sure about this.

How do you suggest we cover these cases?